src/txn/txn_rollback_to_stable.c

```diff
@@ -1191,32 +1185,62 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
         __wt_config_subinit(session, &ckptconf, &cval);
         for (; __wt_config_next(&ckptconf, &key, &cval) == 0;) {
-            ret = __wt_config_subgets(session, &cval, "newest_start_durable_ts", &durableval);
+            ret = __wt_config_subgets(session, &cval, "newest_start_durable_ts", &value);
             if (ret == 0) {
                 newest_start_durable_ts =
-                  WT_MAX(newest_start_durable_ts, (wt_timestamp_t)durableval.val);
+                  WT_MAX(newest_start_durable_ts, (wt_timestamp_t)value.val);
                 durable_ts_found = true;
             }
             WT_ERR_NOTFOUND_OK(ret, false);
-            ret = __wt_config_subgets(session, &cval, "newest_stop_durable_ts", &durableval);
+            ret = __wt_config_subgets(session, &cval, "newest_stop_durable_ts", &value);
             if (ret == 0) {
-                newest_stop_durable_ts =
-                  WT_MAX(newest_stop_durable_ts, (wt_timestamp_t)durableval.val);
+                newest_stop_durable_ts = WT_MAX(newest_stop_durable_ts, (wt_timestamp_t)value.val);
                 durable_ts_found = true;
             }
             WT_ERR_NOTFOUND_OK(ret, false);
-            ret = __wt_config_subgets(session, &cval, "prepare", &durableval);
+            ret = __wt_config_subgets(session, &cval, "prepare", &value);
             if (ret == 0) {
-                if (durableval.val)
+                if (value.val)
                     prepared_updates = true;
             }
             WT_ERR_NOTFOUND_OK(ret, false);
+            ret = __wt_config_subgets(session, &cval, "addr", &value);
+            if (ret == 0)
+                addr_size = value.len;
+            WT_ERR_NOTFOUND_OK(ret, false);
         }
         max_durable_ts = WT_MAX(newest_start_durable_ts, newest_stop_durable_ts);
+
+        /*
+         * The rollback to stable will skip the tables during recovery in the following conditions
+         * 1. Empty table
+         * 2. Table has timestamped updates without a stable timestamp.
+         */
+
+        if (F_ISSET(S2C(session), WT_CONN_RECOVERING) &&
+          (addr_size == 0 ||
+            (txn_global->stable_timestamp == WT_TS_NONE && max_durable_ts != WT_TS_NONE))) {
+            __wt_verbose(session, WT_VERB_RTS, "%s skip rollback to stable because %s", uri,
+              addr_size == 0 ? " checkpoint address length is 0" : "stable timestamp is 0");
+            continue;
+        }
+
+        /* Set this flag to return error instead of panic if file is corrupted. */
+        F_SET(session, WT_SESSION_QUIET_CORRUPT_FILE);
         ret = __wt_session_get_dhandle(session, uri, NULL, NULL, 0);
-        /* Ignore performing rollback to stable on files that don't exist. */
-        if (ret == ENOENT)
+        F_CLR(session, WT_SESSION_QUIET_CORRUPT_FILE);
+
+        /*
+         * Ignore performing rollback to stable on files that don't exist and the files that are
```

Comment:
`don't exist and the` -> `don't exist or the files where the corruption is detected`

Reply:
Fixed